source/agora/consensus/data/Block.d

```diff
@@ -352,13 +352,15 @@ public struct Block
 
 *******************************************************************************/
 
-public Block makeNewBlock (const ref Block prev_block, Transaction[] txs) @safe
+public Block makeNewBlock (Transactions)(const ref Block prev_block,
+    Transactions txs) @safe
+    if (isInputRange!Transactions)
 {
     Block block;
 
     block.header.prev_block = prev_block.header.hashFull();
     block.header.height = prev_block.header.height + 1;
-    block.txs ~= txs;
+    txs.each!(tx => block.txs ~= tx);
```

Comment:
That's way less efficient though

Reply:
Hmm, I didn't think of that. I could add a static if case though.

Reply:
I usually check for the presence of `length`, then do a `reserve`.